lib/index.js

```diff
@@ -37,7 +46,45 @@ class GoodLoggly extends Stream.Writable {
 
     data.timestamp = GoodLoggly.timeString(data.timestamp);
     data.msg = GoodLoggly.getMessage(data);
-    this._client.log(data, data.tags, callback);
+
+    this._buffer.push(data);
+
+    if (this._bufferReady) {
+      this._sendMessages((err) => {
+        this._bufferStart = null;
```

Comment:
This reset logic could be included in `_sendMessages()`, and you could just pass `callback` directly to `_sendMessages()`.